### PR TITLE
Remove uses of shell cd

### LINKED ISSUE #5891: Unable to lint lib or push pod with private repository
* [x] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report
I developing private pods with private repo. Before push I use next command to validate pod
`pod lib lint --verbose --sources=git@bitbucket.org:*/*-ios.git,master --use-libraries --allow-warnings`
Everything was great, but now I getting next error
```
Cloning spec repo `bitbucket-*-ios-1` from `git@bitbucket.*/*-ios.git`
  $ /Applications/Xcode.app/Contents/Developer/usr/bin/git clone git@bitbucket.*/*-ios.git bitbucket-*-ios-1
  Cloning into 'bitbucket-*-ios-1'...
 -> PrivatePhotoLibrary (1.4.10)
    - WARN  | source: Git SSH URLs will NOT work for people behind firewalls configured to only allow HTTP, therefore HTTPS is preferred.
    - ERROR | [iOS] unknown: Encountered an unknown error (Unable to create a source with URL git@bitbucket.org:*/*-ios.git
/Users/romantruba/.rvm/gems/ruby-2.3.0@railstutorial_rails_4_0/gems/cocoapods-1.0.1/lib/cocoapods/sources_manager.rb:42:in `find_or_create_source_with_url'
...
```
After the error, I have a directory "bitbucket-*-ios-1" with private repo content.

Also I'm unable to push into this private repo using `pod push` because of same error.

Checked on two pods. 

## What did you do?

`pod lib lint --verbose --sources=git@bitbucket.org:*/*-ios.git,master --use-libraries --allow-warnings`

## What did you expect to happen?

Lint notifies me about success or code issue.

## What happened instead?

Getting unknown error without any clues what I suppose to do
`ERROR | [iOS] unknown: Encountered an unknown error (Unable to create a source with URL git@bitbucket.org:*/*-ios.git
...`

### Stack

```
   CocoaPods : 1.0.1
        Ruby : ruby 2.3.0p0 (2015-12-25 revision 53290) [x86_64-darwin15]
    RubyGems : 2.6.4
        Host : Mac OS X 10.11.6 (15G1004)
       Xcode : 8.0 (8A218a)
         Git : git version 2.9.3
Ruby lib dir : /Users/romantruba/.rvm/rubies/ruby-2.3.0/lib
Repositories : bitbucket-*-ios - git@bitbucket.org:*/*-ios.git @ 1c511543e32449a9d951fbae87b317ca6fb0d96e
               brion - https://github.com/brion/OGVKit-Specs.git @ 91d779a78c12ae11dc7e601d3124ef848e056f0a
               master - https://github.com/CocoaPods/Specs.git @ a428e56d249fa4f6fae113ef4e7007f17135d39a
               * - git@bitbucket.org:*/*-ios.git @ 3ea4a4a7b39757fed0f659d18832be04ae4c4800
```

### Installation Source

```
Executable Path: /Users/romantruba/.rvm/gems/ruby-2.3.0@railstutorial_rails_4_0/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.0
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.0.0
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
platform :ios, '8.0'
use_frameworks!

source 'git@bitbucket.org:*/*-ios.git'
source 'https://github.com/CocoaPods/Specs.git'

target 'PhotoLibraryTestApplication' do
    pod 'OCMock', '~> 3.3'
    pod 'TwitterLoggingService/ObjC', '~> 2.3'
	pod 'ObjectiveSugar'
    pod 'pop', '~> 1.0'
    pod 'GPUImagePrivate', '~> 0.1'
    pod 'PrivateUIKit', '~> 1.0'
    pod 'RSFoundationUtils', '~> 0.3'
end
```

Thanks.

### PR DESCRIPTION
**Pull request by benasher44**
In most cases, I just switched to passing full paths. For the `realpath` function, I'm 99% we don't need it. In its script, it's used for `PODS_ROOT`, which is always a full path, and `RESOURCE_PATH`, which is checked early on and also set to a full path (relative to `PODS_ROOT`) if it's not one.

### CODE REVIEW

**Comment by DanToml** [comment]
I'm not aware of any cases where it would be set to a symlink or similar, so this seems fine to me.

**Comment by benasher44** [comment]
Looks like @b-ray found the real fix [here](https://github.com/b-ray/CocoaPods/commit/a9ec196dcb9c7757e9709864acfb99cb366d8476). Removing `cd`'s though still feels like a general improvement. I'll update the CHANGELOG and finish updating the tests.

**Comment by DanToml** [comment]
Definitely a general overall improvement!

File: Rakefile [hunk 1]
```diff
@@ -178,7 +178,8 @@ begin
       task :rebuild => :check_for_pending_changes do
         tarballs.each do |tarball|
           basename = File.basename(tarball)
-          sh "cd #{File.dirname(tarball)} && rm #{basename} && env COPYFILE_DISABLE=1 tar -zcf #{basename} #{basename[0..-8]}"
+          untarred_path = File.join(File.dirname(tarball), basename[0..-8])
+          sh "rm #{tarball} && env COPYFILE_DISABLE=1 tar -zcf #{tarball} #{untarred_path}"
```
Comment (DanToml) [inline Rakefile]: 👍 so much easier to read too. Nice!

**Pull request merged**